docker/resource_docker_container_v1.go

```diff
@@ -0,0 +1,777 @@
+package docker
+
+import "github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+
+func resourceDockerContainerV1() *schema.Resource {
+	return &schema.Resource{
+		//This is only used for state migration, so the CRUD
+		//callbacks are no longer relevant
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"rm": {
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+			},
+
+			"read_only": {
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"start": {
+				Type:     schema.TypeBool,
+				Default:  true,
+				Optional: true,
+			},
+
+			"attach": {
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+			},
+
+			"logs": {
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+			},
+
+			// Indicates whether the container must be running.
+			//
+			// An assumption is made that configured containers
+			// should be running; if not, they should not be in
+			// the configuration. Therefore a stopped container
+			// should be started. Set to false to have the
+			// provider leave the container alone.
+			//
+			// Actively-debugged containers are likely to be
+			// stopped and started manually, and Docker has
+			// some provisions for restarting containers that
+			// stop. The utility here comes from the fact that
+			// this will delete and re-create the container
+			// following the principle that the containers
+			// should be pristine when started.
+			"must_run": {
+				Type:     schema.TypeBool,
+				Default:  true,
+				Optional: true,
+			},
+
+			"exit_code": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"container_logs": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			// ForceNew is not true for image because we need to
+			// sane this against Docker image IDs, as each image
+			// can have multiple names/tags attached do it.
+			"image": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
It looks like this will still force replacement on all docker containers since the images won't match. Why `ForceNew: true` here?

Reply:
This isn't being changed by this pull request.
`ForceNew: true` is set in the master branch too.

https://github.com/terraform-providers/terraform-provider-docker/blob/7803a1fd929e70143d70e30bba1956401aae1b01/docker/resource_docker_container.go#L884

I don't know the reason, but `ForceNew: true` is conflicting the code comment `ForceNew is not true for image`.
It looks strange.

Reply:
Not only `resourceDockerContainerV1` but also `resourceDockerContainer` look strange.

https://github.com/terraform-providers/terraform-provider-docker/blob/7803a1fd929e70143d70e30bba1956401aae1b01/docker/resource_docker_container.go#L103-L109

Reply:
Hmm. I started using terraform after the commit that always forced replacement, so I've never known it to work. But I don't know how it would ever work that containers aren't overwritten. Unless maybe `ForceNew` means "force replacement only if this value changes, not if the hash changes"?

I checked and there has been a mismatch in the comment since the initial commit 6 years ago: https://github.com/terraform-providers/terraform-provider-docker/commit/77e1e6067e43b4f435ed3d285ed277cf2dbc33ec#diff-79b16c3fa28649a2f193d574756c3558R46-R53

Reply:
Oh, over 5 years ago...
https://github.com/terraform-providers/terraform-provider-docker/commit/77e1e6067e43b4f435ed3d285ed277cf2dbc33ec#diff-79b16c3fa28649a2f193d574756c3558R46-R52

Reply:
https://www.terraform.io/docs/providers/docker/r/container.html#image

> image - (Required, string) The ID of the image to back this container.
> The easiest way to get this value is to use the docker_image resource as is shown in the example above.

The value of `image` is Docker image ID, and if the Docker image ID is changed the container should be recreated.
So I think `ForceNew: true` is correct, and the code comment is wrong.

Reply:
I want to fix the code comment in the other pull request, because the objective of this pull request is different from it.